segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/NoOpStorageFactory.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.noop;
+
+import com.google.common.base.Preconditions;
+import io.pravega.segmentstore.storage.AsyncStorageWrapper;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.segmentstore.storage.rolling.RollingStorage;
+
+import java.util.concurrent.Executor;
+
+/**
+ * Factory for No-Op mode Storage adapters.
+ */
+public class NoOpStorageFactory implements StorageFactory {
+    private final StorageExtraConfig config;
+    private final Executor executor;
+    private final StorageFactory systemStorageFactory;
+    private final StorageFactory userStorageFactory;
+
+    public NoOpStorageFactory(StorageExtraConfig config, Executor executor, StorageFactory systemStorageFactory, StorageFactory userStorageFactory) {
```

Comment:
When is `userStorageFactory` ever non-null? I see in `StorageLoader` that you pass `null` as this argument.

Reply:
In the unit test of NoOpStorage itself, the supposedly no-oped segments are stored in this storage.
It is essential to verify the no-oped segment to ensure the no-op storage works as expected. For example, there is a rollingStorage layer inbetween NoOpStorage and underlying base storage, rolling layer may seal segment when things got all "rolled" up. If we don't verify the No-Oped segment, then it's impossible to test whether the rolling layer is doing things correctly.